frontend/javascripts/oxalis/view/left-border-tabs/modals/add_volume_layer_modal.tsx

```diff
@@ -17,11 +17,67 @@ import {
 import {
   getAllReadableLayerNames,
   getVolumeTracingLayers,
-  validateReadableLayerName,
 } from "oxalis/model/accessors/volumetracing_accessor";
+import messages from "messages";
 import InputComponent from "oxalis/view/components/input_component";
 import api from "oxalis/api/internal_api";
 import Toast from "libs/toast";
+
+export type ValidationResult = { isValid: boolean; message: string };
+export function checkForLayerNameDuplication(
+  readableLayerName: string,
+  allReadableLayerNames: string[],
+): ValidationResult {
+  const layerNameDoesNotExist = _.countBy(allReadableLayerNames)[readableLayerName] <= 0;
+  return {
+    isValid: layerNameDoesNotExist,
+    message: layerNameDoesNotExist ? "" : messages["tracing.volume_layer_name_duplication"],
+  };
+}
+
+export function checkLayerNameForInvalidCharacters(readableLayerName: string): ValidationResult {
+  const uriSaveCharactersRegex = /[0-9a-zA-Z-._~]+/g;
```

Comment:
I also like this version much more than the brute-force one. As you said, it's easier to understand and more elegant. I'm certain that the list of safe characters is correct, see [this post](https://stackoverflow.com/a/695469) which references the corresponding RFC section.